api/portainer.go

```diff
@@ -976,7 +992,7 @@ type (
 
 	// OAuthService represents a service used to authenticate users using OAuth
 	OAuthService interface {
-		Authenticate(code string, configuration *OAuthSettings) (string, error)
+		Authenticate(code string, configuration *OAuthSettings) (OAuthUserData, error)
```

Comment:
Another change of a public API. More and more I get the feeling that this is a rather broadbanded public-ication of interfaces. Weird that implementation choices are missing.

That in hindsight I suggest to keep going and leave the changes as is. I fail to provide any better alternative for now.